### ISSUE TITLE
PHPStan reporting an error for missing optional parameters

### ISSUE THREAD

**Issue by ktrzos** [loophp/collection#272]
Hey 👋!

I have this code:

```php
$this->commandList
	->filter(static fn (CommandInterface $command) => $command instanceof $commandClass)
	->count();
```

and I'm receiving such PHPStan error since upgrading PHPStan to 1.8.* (now using 1.9.*):

> Parameter ...$callbacks of method loophp\collection\Contract\Operation\Filterable<int,CommandInterface>::filter() expects
> callable(CommandInterface=, int=, Iterator<int, CommandInterface>=): bool, 
> Closure(CommandInterface): bool given.

How to manage with this problem? I'm using v6.0.3 of `loophp/collection`. Thank you in advance for any help.

**Comment by drupol** [comment]
Hello, 

I'm also having the issue, I guess this is something in PHPStan.

I'll investigate a bit today and give you more info very soon.

Do you have some code to reproduce the issue easily ?

**Comment by ktrzos** [comment]
I think I've replicated it in PhpStan's sandbox:
https://phpstan.org/r/e9b90c05-c621-449a-a555-ee58d3c35328

**Comment by drupol** [comment]
Do you think it would be possible to create a smaller reproducer on phpstan.org with a smaller Collection footprint?

**Comment by ktrzos** [comment]
Yes: https://phpstan.org/r/a89282ef-e9e9-4398-8f20-d8e2bafaa60a
The `Collection` interface is just a "copy" of the code from the `Filterable`.

**Comment by drupol** [comment]
Thanks ! Going to open a bug report at PHPStan.

**Comment by drupol** [comment]
Dear @ktrzos,

I fixed this issue in 6a2755101486841531d2e60f25895c24823a6490

Thanks for reporting it :)

I also fixed your snippet: https://phpstan.org/r/057f016d-c7a4-48ff-beea-a14d0d27bd71

All of this will be available in upcoming version 7.
